src/System.Management.Automation/utils/tracing/SysLogProvider.cs

```diff
@@ -0,0 +1,504 @@
+    ///
+    ///  Event
+    ///   Application logging (Events)
+    ///   Context: EventId:taskname.opcodename.levelname
+    ///   Payload: The event's message text formatted with arguments from the caller.
+    ///   Example: (19E1025:3:10) [Perftrack_ConsoleStartupStart:PowershellConsoleStartup.WinStart.Informational] PowerShell console is starting up
+    /// </remarks>
+    internal class SysLogProvider
+    {
+        // The full git commit id string
+        static readonly string _gitCommitId;
+        // The hash code for the git commit id.
+        static readonly int _commitId;
+
+        [ThreadStatic]
+        static Guid _activity = Guid.NewGuid();
+
+        // Ensure the string pointer is not garbage collected.
+        static IntPtr _nativeSyslogIdent = IntPtr.Zero;
+        static NativeMethods.SysLogPriority _facility = NativeMethods.SysLogPriority.Local0;
+
+        byte _channelFilter;
+        ulong _keywordFilter;
+        byte _levelFilter;
+
+        static SysLogProvider()
+        {
+            // full git commit id string. Will be logged with the associated hash code.
+            _gitCommitId = PSVersionInfo.GitCommitId;
+            // has value for the commit id string.
+            _commitId = _gitCommitId.GetHashCode();
+        }
+
+        /// <summary>
+        /// Initializes a new instance of this class
+        /// </summary>
+        /// <param name="applicationId">The log identity name used to identify the application in syslog.</param>
+        /// <param name="level">The trace lavel to enable.</param>
+        /// <param name="keywords">The keywords to enable.</param>
+        /// <param name="channels">The output channels to enable.</param>
+        public SysLogProvider(string applicationId, PSLevel level, PSKeyword keywords, PSChannel channels)
+        {
+            // NOTE: This string needs to remain valid for the life of the process since the underlying API keeps
+            // a reference to it.
+            // FUTURE: If logging is redesigned, make these details static or a singleton since there should only be one
+            // instance active.
+            _nativeSyslogIdent = Marshal.StringToHGlobalAnsi(applicationId);
+            NativeMethods.OpenLog(_nativeSyslogIdent, _facility);
+            _keywordFilter = (ulong)keywords;
+            _levelFilter = (byte) level;
+            _channelFilter = (byte) channels;
+            LogCommit();
+        }
+
+        /// <summary>
+        /// Gets the value indicating if the specified level and keywords are enabled for logging
+        /// </summary>
+        /// <param name="level">The PSLevel to check.</param>
+        /// <param name="keywords">The PSKeyword to check.</param>
+        /// <returns>true if the specified level and keywords are enabled for logging.</returns>
+        internal bool IsEnabled(PSLevel level, PSKeyword keywords)
+        {
+            return
```

Comment:
We avoid unnecessary lines in this code base, this should be 1 or 2 lines instead of 6.

Reply:
Fixed